connector/src/main/java/org/geysermc/connector/network/translators/java/JavaPluginMessageTranslator.java

```diff
@@ -28,53 +28,91 @@
 import com.github.steveice10.mc.protocol.packet.ingame.client.ClientPluginMessagePacket;
 import com.github.steveice10.mc.protocol.packet.ingame.server.ServerPluginMessagePacket;
 import com.google.common.base.Charsets;
+import com.nukkitx.protocol.bedrock.packet.EmotePacket;
 import org.geysermc.connector.common.AuthType;
+import org.geysermc.connector.entity.Entity;
 import org.geysermc.connector.network.session.GeyserSession;
 import org.geysermc.connector.network.translators.PacketTranslator;
 import org.geysermc.connector.network.translators.Translator;
+import org.geysermc.connector.utils.PluginMessageUtils;
+import org.geysermc.connector.utils.StringByteUtil;
 import org.geysermc.cumulus.Form;
 import org.geysermc.cumulus.Forms;
 import org.geysermc.cumulus.util.FormType;
 
+import java.nio.ByteBuffer;
 import java.nio.charset.StandardCharsets;
 
 @Translator(packet = ServerPluginMessagePacket.class)
 public class JavaPluginMessageTranslator extends PacketTranslator<ServerPluginMessagePacket> {
     @Override
     public void translate(GeyserSession session, ServerPluginMessagePacket packet) {
-        // The only plugin messages it has to listen for are Floodgate plugin messages
-        if (session.getRemoteAuthType() != AuthType.FLOODGATE) {
-            return;
+        // Handle plugin channels
+        switch (packet.getChannel()) {
```

Comment:
Use java 17 feature with lambda

Reply:
Do note that Geyser is not yet on Java 17.

Reply:
Switch pattern matching is a preview feature on J17. Why would I use it?